cpp/src/io/parquet/reader_impl_preprocess.cu

```diff
@@ -254,25 +269,63 @@ template <typename T = uint8_t>
       next_chunk++;
     }
     if (io_size != 0) {
-      auto& source = sources[chunk_source_map[chunk]];
-      if (source->is_device_read_preferred(io_size)) {
-        auto buffer        = rmm::device_buffer(io_size, stream);
-        auto fut_read_size = source->device_read_async(
-          io_offset, io_size, static_cast<uint8_t*>(buffer.data()), stream);
+      auto& source              = sources[chunk_source_map[chunk]];
```

Comment:
I am guessing that `is_device_read_preferred` is `const`, which would make this const-friendly.
```suggestion
      auto const& source        = sources[chunk_source_map[chunk]];
```